solr_helpers/__init__.py

```diff
@@ -126,7 +126,8 @@ def query_solr(collection=None, fields=None, query_string=None, fqs=None, facets
             payload['facet']['unique_{}'.format(x)] = {
                 'type': 'terms',
                 'field': ufield,
-                'limit': -1,
+                'offset': offset,
+                'limit': limit,
                 'missing': True,
```

Comment:
As above: don't apply a limit to facets, or it will only give you counts for that many records, which isn't what we want.